### ISSUE TITLE
Attempt to upload a file greater than the server file size limit using http_upload fails with no feedback in the GUI

### ISSUE THREAD

**Issue by sam-hobbs** [siacs/Conversations#2533]
#### General information

* **Version:** 1.18.5
* **Device:** Fairphone 2
* **Android Version:** Fairphone Open Source OS / Android 6.0.1
* **Server name:** self hosted
* **Server software:** Prosody version 0.10 nightly build 402 (2017-06-14, 39188851811c)
* **Installed server modules:** MAM, MUC, http_upload
* **Conversations source:** F-Droid

#### Steps to reproduce

1. Set up prosody server with MUC room & more than one participant (to force file sharing via http_upload)
2. Attempt to upload a large file (e.g. video) to the chat, which has a filesize greater than the server limit
3. Conversations will compress the video successfully, and then attempt to upload it. User sees an entry with an "OPEN VIDEO" button, and a label underneath that says "waiting...".

#### Expected result

The user should see a message that the file is too big to upload, such as "delivery failed: file size exceeds server limit"

_and/or_

conversations should prevent the user from attempting to upload files that are too big in the first place (depending on when it retrieves the server file size limit) - could the file picker be filtered by file size?

Conversations knows that the file is too large because a message appears in the log (see below)

#### Debug output

````
me@mydomain.com: http upload is not available for files with size 10879603 (max is 1048576)
````

I was looking at the server logs at first and couldn't find anything related to the upload attempt (not even debug level) so I think conversations doesn't even attempt to upload the file because it knows it is too big.

**Comment by licaon-kter** [comment]
Not fixed since #2482?

**Comment by sam-hobbs** [comment]
@licaon-kter Nope - I can see that #2482 helps in one on one chats where you want to transfer a large file (bigger than the http_upload limit) because it won't try and fall back to Jingle. Still leaves conversations in limbo if the file exceeds the limit and it's a group chat, though.

@iNPUTmice that was quick! Got to it before I could even type a reply. Thanks.
